Adding agent in conversation counter in QueueSummary AMI event (based on AST_DEVICE_INUSE flag)
Index: asterisk-20.2.1/apps/app_queue.c
===================================================================
--- asterisk-20.2.1.orig/apps/app_queue.c
+++ asterisk-20.2.1/apps/app_queue.c
@@ -10334,6 +10334,7 @@ static int manager_queues_summary(struct
 {
 	time_t now;
 	int qmemcount = 0;
+	int qmembusy = 0;
 	int qmemavail = 0;
 	int qchancount = 0;
 	int qlongestholdtime = 0;
@@ -10365,6 +10366,7 @@ static int manager_queues_summary(struct
 		if (ast_strlen_zero(queuefilter) || !strcasecmp(q->name, queuefilter)) {
 			/* Reset the necessary local variables if no queuefilter is set*/
 			qmemcount = 0;
+			qmembusy = 0;
 			qmemavail = 0;
 			qchancount = 0;
 			qlongestholdtime = 0;
@@ -10378,6 +10380,9 @@ static int manager_queues_summary(struct
 						++qmemavail;
 					}
 				}
+				if (mem->status == AST_DEVICE_INUSE) {
+					++qmembusy;
+				}
 				ao2_ref(mem, -1);
 			}
 			ao2_iterator_destroy(&mem_iter);
@@ -10391,13 +10396,14 @@ static int manager_queues_summary(struct
 				"Queue: %s\r\n"
 				"LoggedIn: %d\r\n"
 				"Available: %d\r\n"
+				"Talking: %d\r\n"
 				"Callers: %d\r\n"
 				"HoldTime: %d\r\n"
 				"TalkTime: %d\r\n"
 				"LongestHoldTime: %d\r\n"
 				"%s"
 				"\r\n",
-				q->name, qmemcount, qmemavail, qchancount, q->holdtime, q->talktime, qlongestholdtime, idText);
+				q->name, qmemcount, qmemavail, qmembusy, qchancount, q->holdtime, q->talktime, qlongestholdtime, idText);
 			++qsummaries;
 		}
 		ao2_unlock(q);
